classes/publication/Repository.php

```diff
@@ -767,6 +767,24 @@ protected function getErrorMessageOverrides(): array
         ];
     }
 
+    /**
+     * Assign categories to a publication.
+     *
+     */
+    public function assignCategoriesToPublication(int $publicationId, array $categoryIds): void
+    {
+        // delete all existing entries for the publication
+        PublicationCategory::where('publication_id', $publicationId)->delete();
+
+        // insert new categories if provided
+        foreach ($categoryIds as $categoryId) {
+            PublicationCategory::create([
+                'publication_id' => $publicationId,
+                'category_id' => $categoryId
+            ]);
```

Comment:
I think you can drop the loop and replace the `PublicationCategory::create()` by `PublicationCategory::insert()`, which accepts an array of records, then I guess only a single database query will be issued.

I think it's nice to chunk such mass operations (1000 items), even though it's unlikely this method will receive a lot of items.

Reply:
Hmm, in fact, a better option (safer, as we're not removing everything first, and perhaps more performatic) might be to use the `PublicationCategory::upsert()` to update/insert all the items, then move the `delete()` to the end (`where('publicationId', $publicationId)->whereNotIn('category_id', $categoryIds)`.